### PR TITLE
kfctl: add default zone if not specified

### PR DESCRIPTION
**Pull request by r2d4**
/assign @abhi-g 

I'll send a follow up cherry-pick to v0.4-branch

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2210)

### CODE REVIEW

File: scripts/kfctl.sh [hunk 1]
```diff
@@ -211,9 +211,8 @@ parseArgs() {
     if [ -z "$ZONE" ]; then
       ZONE=$(gcloud config get-value compute/zone 2>/dev/null)
       if [ -z "$ZONE" ]; then
-        echo "GCP zone must be set either using --zone <ZONE>"
-        echo "or by setting a default zone in gcloud config"
-        exit 1
+        # Set default zone to us-east1-d if not specified
+        ZONE="us-east1-d"
```
Comment (abhi-g) [inline scripts/kfctl.sh]: Also echo a message suggesting that we are setting the default to us-east1d. but can be overriden using gcloud config or --zone

Comment (abhi-g) [reply scripts/kfctl.sh]: essentially the previous echo statements could be brought back though not the exit 1 :)

Comment (r2d4) [reply scripts/kfctl.sh]: Yep - made it a bit cleaner.

**Comment by abhi-g** [comment]
fixes #2209 
/lgtm
/approve

**Pull request merged**